.github/workflows/test.yml

```diff
@@ -12,51 +13,111 @@ jobs:
+
+  Unit-Tests:
+    needs: Setup
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Restore cache
         uses: actions/cache@v4
-        id: frontend-cache
         with:
           path: |
             ~/.cache/Cypress
+            ${{ github.workspace }}/node_modules
+            ${{ github.workspace }}/backend/node_modules
             ${{ github.workspace }}/frontend/node_modules
-          key: ${{ runner.os }}-${{ matrix.node-version }}-modules-frontend-${{ hashFiles('**/frontend/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ matrix.node-version }}-modules-frontend
-      - name: Install Node.js packages
-        if: ${{ steps.repo-cache.outputs.cache-hit != 'true' || steps.backend-cache.outputs.cache-hit != 'true' || steps.frontend-cache.outputs.cache-hit != 'true' }}
-        run: npm install
-      - name: Check for uncomitted changes
+          key: ${{ runner.os }}-18.x-modules-${{ hashFiles('**/package-lock.json') }}
+      - name: Run unit tests
+        run: npm run test:unit
+
+  Get-Test-Groups:
+    runs-on: ubuntu-latest
+    outputs:
+      test-groups: ${{ steps.set-groups.outputs.test-groups }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-groups
+        shell: bash
         run: |
-          git diff --exit-code
-      - name: Test backend & test frontend with code coverage
-        if: ${{ success() }}
-        run: npm run test:backend && npm run test:frontend:coverage
-      - name: Upload coverage reports to Codecov
-        uses: codecov/codecov-action@v4.6.0
-        with:
-          token: ${{ secrets.CODECOV_TOKEN }}
-          fail_ci_if_error: false
-          name: frontend
-          directory: ./frontend/coverage
-          verbose: true
-      - uses: actions/upload-artifact@v4
-        if: ${{ always() }}
+          set -x  # Enable debug mode to see each command
+          
+          echo "Checking directory structure..."
+          ls -la frontend/src/__tests__/cypress/cypress/tests/ || echo "Base test directory not found"
+          
+          if [ -d "frontend/src/__tests__/cypress/cypress/tests/mocked" ]; then
+            echo "Found mocked tests directory"
+            
+            # Get directories and create JSON array - force compact output with -c
+            DIRS=$(cd frontend/src/__tests__/cypress/cypress/tests/mocked && \
+                  find . -maxdepth 1 -mindepth 1 -type d -printf "%f\n" | \
+                  jq -R . | jq -c -s .)
+            
+            # Output in the correct format for GitHub Actions
+            echo "test-groups=$DIRS" >> "$GITHUB_OUTPUT"
+            
+            echo "Generated test groups:"
+            cat "$GITHUB_OUTPUT"
+          else
+            echo "No mocked tests directory found, using default"
+            echo "test-groups=[\"default\"]" >> "$GITHUB_OUTPUT"
+          fi
+
+  Cypress-Tests:
+    needs: [Setup, Get-Test-Groups]
```

Comment:
Should we also require the Unit-Tests to pass successfully in order to execute the Mocks? If any of the Unit tests there is a possibility that the mocks would also fail.